app/helpers/sessions_helper.rb

```diff
@@ -38,7 +39,30 @@ def clear_cart
     end
     
     def log_out
+        user = User.find_by(id: session[:user_id])
+        user.update_attribute(:cart, hto_s(session[:cart]))
```

Comment:
@modsoussi Note that `update_attribute` skips validations (see handy blog post on [updating models](http://www.davidverhasselt.com/set-attributes-in-activerecord/)).